ste/xfer-anyToRemote.go

```diff
@@ -141,6 +141,18 @@ func anyToRemote(jptm IJobPartTransferMgr, p pipeline.Pipeline, pacer pacer, sen
 		}
 	}
 
+	// step 5a: lock the destination
+	// (is safe to do it relatively early here, before we run the prologue, because its just a internal lock, within the app)
+	// But must be after all of the early returns that are above here (since
+	// if we succeed here, we need to know the epilogue will definitely run to unlock later)
+	err = jptm.WaitUntilLockDestination(jptm.Context())
```

Comment:
>WaitUntilLockDestination [](start = 12, length = 24)

do we want to avoid the overhead of locking if we are targeting blob/ADLS Gen2?

Reply:
Good question, and yes, those would be the only two times we can skip it. (All downloads need it, for the --decompress renamings, and all sending to Azure Files needs it).  How about we consider avoiding it in those cases in the next release? (Need to think about pros/cons of maintainabilyt and complexity of turning it on and off).  I've made a backlog item.